File: sportsbook-app/payment-contracts/src/main/java/com/sportsbook/payment/contracts/DepositValidatedEvent.java
package com.sportsbook.payment.contracts;

import java.math.BigDecimal;
import java.time.Instant;

public record DepositValidatedEvent(
        String depositId,
        String playerId,
        BigDecimal amount,
        FundingSourceType fundingSourceType,
        DepositDecision decision,
        DepositRejectionReason rejectionReason,
        String providerReference,
        Instant decidedAt
) {}

File: sportsbook-app/payment-contracts/src/main/kotlin/com/sportsbook/payment/contracts/DepositValidatedEvent.kt
package com.sportsbook.payment.contracts

import java.math.BigDecimal
import java.time.Instant

data class DepositValidatedEvent(
    val depositId: String,
    val playerId: String,
    val amount: BigDecimal,
    val fundingSourceType: FundingSourceType,
    val decision: DepositDecision,
    val rejectionReason: DepositRejectionReason?,
    val providerReference: String?,
    val decidedAt: Instant,
)
